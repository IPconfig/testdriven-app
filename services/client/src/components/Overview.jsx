import React from 'react';

const Overview = (props) => {
    return (
        <div className="column has-text-centered">
        <div className="reactor">
        
        {
            props.tube_states.map((row, index) => {
                return (
                    <div key={index} className="reactor-row">
                    {
                        row.map((element, idx) => {
                            return (
                                <div key={idx} className={`tube state${ element }`}>
                                   {/* { element }  values */ }
                                </div>
                        )})
                    }
                    </div>
                )
            })
        }
        </div>
        </div>
    )
};

export default Overview;