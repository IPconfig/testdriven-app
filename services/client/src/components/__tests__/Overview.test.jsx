import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Overview from '../Overview';

const tube_states = [
    [2,3],
    [1,1,1,1],
    [2,2,2,2],
    [3,3,3,3],
    [4,4,4,4],
    [5,5,5,5],
    [0,4]
];

test('Overview renders properly', () => {
    const wrapper = shallow(<Overview tube_states={tube_states}/>);
 //   expect(wrapper.find('h1').get(0).props.children).toBe('Reactor Overview');
    // reactor
    const reactor = wrapper.find('.reactor');
    expect(reactor.length).toBe(1);
    // reactor rows
    const rows = wrapper.find('.reactor-row');
    expect(rows.length).toBe(7);
    // tubes
    expect(rows.get(0).props.children.length).toBe(2);
    expect(rows.get(1).props.children.length).toBe(4);
    expect(rows.get(2).props.children.length).toBe(4);
    expect(rows.get(3).props.children.length).toBe(4);
    expect(rows.get(4).props.children.length).toBe(4);
    expect(rows.get(5).props.children.length).toBe(4);
    expect(rows.get(6).props.children.length).toBe(2);
});

test('Overview renders a snapshot properly', () => {
    const tree = renderer.create(<Overview tube_states={tube_states}/>).toJSON();
    expect(tree).toMatchSnapshot();
})