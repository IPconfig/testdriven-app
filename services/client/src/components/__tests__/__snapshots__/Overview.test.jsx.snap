// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders a snapshot properly 1`] = `
<div
  className="column has-text-centered"
>
  <div
    className="reactor"
  >
    <div
      className="reactor-row"
    >
      <div
        className="tube state2"
      />
      <div
        className="tube state3"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state1"
      />
      <div
        className="tube state1"
      />
      <div
        className="tube state1"
      />
      <div
        className="tube state1"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state2"
      />
      <div
        className="tube state2"
      />
      <div
        className="tube state2"
      />
      <div
        className="tube state2"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state3"
      />
      <div
        className="tube state3"
      />
      <div
        className="tube state3"
      />
      <div
        className="tube state3"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state4"
      />
      <div
        className="tube state4"
      />
      <div
        className="tube state4"
      />
      <div
        className="tube state4"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state5"
      />
      <div
        className="tube state5"
      />
      <div
        className="tube state5"
      />
      <div
        className="tube state5"
      />
    </div>
    <div
      className="reactor-row"
    >
      <div
        className="tube state0"
      />
      <div
        className="tube state4"
      />
    </div>
  </div>
</div>
`;
